### ISSUE TITLE
LicenseCandidates:Error in merging/verifiying-Prozess

### ISSUE THREAD

**Issue by NesrinCeylan** [fossology/fossology#938]
In the course of inserting a new license I received the following error on cli: 

fossology-db_1          | LOG:  could not receive data from client: Connection reset by peer
fossology-db_1          | ERROR:  duplicate key value violates unique constraint "rf_md5unique"
fossology-db_1          | DETAIL:  Key (rf_md5)=(36ef664ecb655e91bf32c72773ce05d5) already exists.
fossology-db_1          | STATEMENT:  INSERT INTO license_ref (rf_pk, rf_shortname, rf_text, rf_url, rf_add_date, rf_copyleft,
fossology-db_1          | 	        "rf_OSIapproved", rf_fullname, "rf_FSFfree", "rf_GPLv2compatible", "rf_GPLv3compatible", rf_notes, "rf_Fedora",
fossology-db_1          | 	        marydone, rf_active, rf_text_updatable, rf_md5 , rf_detector_type, rf_risk)
fossology-db_1          | 	      (SELECT rf_pk, $2 as rf_shortname, rf_text, rf_url, now() as rf_add_date, rf_copyleft,
fossology-db_1          | 	        "rf_OSIapproved", rf_fullname, "rf_FSFfree", "rf_GPLv2compatible", "rf_GPLv3compatible", rf_notes, "rf_Fedora",
fossology-db_1          | 	        false AS marydone, rf_active, rf_text_updatable, md5(rf_text) rf_md5 , 1 rf_detector_type, rf_risk
fossology-db_1          | 	  FROM license_candidate WHERE rf_pk=$1)
fossology-web_1         | [Thu Nov 09 07:38:04.622045 2017] [:error] [pid 19] [client 172.18.0.1:52126] PHP Warning:  pg_execute(): Query failed: ERROR:  duplicate key value violates unique constraint &quot;rf_md5unique&quot;\nDETAIL:  Key (rf_md5)=(36ef664ecb655e91bf32c72773ce05d5) already exists. in /usr/local/share/fossology/lib/php/Db/Driver/Postgres.php on line 72, referer: http://localhost:8081/repo/?mod=admin_license_candidate&rf=601
fossology-web_1         | [Thu Nov 09 07:38:04.622247 2017] [:error] [pid 19] [client 172.18.0.1:52126] [2017-11-09 07:38:04] default.CRITICAL: ERROR:  duplicate key value violates unique constraint "rf_md5unique" DETAIL:  Key (rf_md5)=(36ef664ecb655e91bf32c72773ce05d5) already exists. [] [], referer: http://localhost:8081/repo/?mod=admin_license_candidate&rf=601

Usually I use the WebUI working on fossology. I'm trying to test all functionallity on UI to learn, how fossology work.  

Where is my mistake. Sometimes it works, and sometimes not. 
As I understood, the process of adding a new license is to create new license on "organize -> Licenses -> new license". After adding I turn to the page "Admin -> License Admin -> Candidates"  and trying to verify or merge or compare. 

What I also ask myself is whether it's the right thing to do, or whether I have to set up adding somewhere else the license. My goal is not only to add new license, but also to replace, for instance "unknown license". Any Idea? 
Thanks in advance

**Comment by steffen-weber** [comment]
Looks like you want to add two licenses with identical texts.

**Comment by mcjaeger** [comment]
actually observation is still there with 3.4:

```
[Fri Feb 15 19:21:40.681616 2019] [:error] [pid 13257] [client 10.0.2.2:56782] PHP Warning:  pg_execute(): Query failed: ERROR:  duplicate key value violates unique constraint &quot;rf_md5unique&quot;\nDETAIL:  Key (rf_md5)=(ca3b791701f63f869d4a0dc30812e64b) already exists. in /usr/share/fossology/lib/php/Db/Driver/Postgres.php on line 72, referer: http://localhost:8081/repo/?mod=admin_license_candidate&rf=726
[Fri Feb 15 19:21:40.682292 2019] [:error] [pid 13257] [client 10.0.2.2:56782] [2019-02-15 19:21:40] default.CRITICAL: ERROR:  duplicate key value violates unique constraint "rf_md5unique" DETAIL:  Key (rf_md5)=(ca3b791701f63f869d4a0dc30812e64b) already exists. [] [], referer: http://localhost:8081/repo/?mod=admin_license_candidate&rf=726
[Fri Feb 15 19:21:40.691626 2019] [:error] [pid 13257] [client 10.0.2.2:56782] PHP Fatal error:  Uncaught Fossology\\Lib\\Exception: error executing: Fossology\\UI\\Page\\AdminLicenseCandidate::verifyCandidate.insert: INSERT INTO license_ref (rf_pk, rf_shortname, rf_text, rf_url, rf_add_date, rf_copyleft,\n        "rf_OSIapproved", rf_fullname, "rf_FSFfree", "rf_GPLv2compatible", "rf_GPLv3compatible", rf_notes, "rf_Fedora",\n        marydone, rf_active, rf_text_updatable, rf_md5 , rf_detector_type, rf_risk)\n      (SELECT rf_pk, $2 as rf_shortname, rf_text, rf_url, now() as rf_add_date, rf_copyleft,\n        "rf_OSIapproved", rf_fullname, "rf_FSFfree", "rf_GPLv2compatible", "rf_GPLv3compatible", rf_notes, "rf_Fedora",\n        false AS marydone, rf_active, rf_text_updatable, md5(rf_text) rf_md5 , 1 rf_detector_type, rf_risk\n  FROM license_candidate WHERE rf_pk=$1) -- -- Array\n(\n    [0] => 726\n    [1] => sodor\n)\n\n\nERROR:  duplicate key value violates unique constraint "rf_md5unique"\nDETAIL:  Key (rf_md5)=(ca3b791701f63f869d4a0dc30812e64b) already exists. in /usr/share/fossology/lib/php/Db/DbManager.php:140\nStack tr in /usr/share/fossology/lib/php/Db/DbManager.php on line 140, referer: http://localhost:8081/repo/?mod=admin_license_candidate&rf=726
```

adding a license with the same text returns white page to the user and dumps the log above to `/var/log/apache2/error.log`

**Comment by adityabisoi** [comment]
@mcjaeger this issue has been solved in MR #1613
